### PR TITLE
feat(ontology): edit res class info (DSP-1209)

### PR DESCRIPTION
**Pull request by kilchenmann**
resolves DSP-1209

### CODE REVIEW

File: src/app/project/ontology/resource-class-form/resource-class-form.component.ts [hunk 1]
```diff
@@ -344,9 +367,58 @@ export class ResourceClassFormComponent implements OnInit, OnDestroy, AfterViewC
...
         this.loading = true;
         if (this.edit) {
-            // edit mode
-            // submit properties and set cardinality
-            this.submitProps(this.resourceClassForm.value.properties, this.iri);
+
+            if (this.showResourceClassForm) {
+                // edit mode: res class info (label and comment)
+                // label
+                const onto4Label = new UpdateOntology<UpdateResourceClassLabel>();
```
Comment (mdelez) [inline src/app/project/ontology/resource-class-form/resource-class-form.component.ts:374]: I'm not sure I understand this variable name 😅 same with `onto4comment`

File: src/app/project/ontology/resource-class-form/resource-class-form.component.ts [hunk 2]
```diff
@@ -344,9 +367,58 @@ export class ResourceClassFormComponent implements OnInit, OnDestroy, AfterViewC
...
+                updateComment.comments = this.resourceClassComments;
+                onto4Comment.entity = updateComment;
+
+                this._dspApiConnection.v2.onto.updateResourceClass(onto4Label).subscribe(
+                    (classResponse: ResourceClassDefinitionWithAllLanguages) => {
+                        this.lastModificationDate = classResponse.lastModificationDate;
+                        onto4Comment.lastModificationDate = this.lastModificationDate;
+
+                        this._dspApiConnection.v2.onto.updateResourceClass(onto4Comment).subscribe(
+                            (classResponse: ResourceClassDefinitionWithAllLanguages) => {
```
Comment (mdelez) [inline src/app/project/ontology/resource-class-form/resource-class-form.component.ts:398]: classResponse is shadowed here but it's not a big issue

**Review by mdelez — CHANGES_REQUESTED** [review]
Looks pretty good! `ResourceClassForm` could use a little clean up to remove unused parameters and add a missing semicolon on [line 438](https://github.com/dasch-swiss/dsp-app/blob/d7cb2d4c83cd41176eb71e4a2d4bde847272086d/src/app/project/ontology/resource-class-form/resource-class-form.component.ts#L438). Also you don't appear to set `this.loading` to false in the `SubmitData` method if the user is *not* in edit mode.

File: src/app/project/ontology/resource-class-form/resource-class-form.component.ts (hunk 1 above)
Comment (kilchenmann) [reply src/app/project/ontology/resource-class-form/resource-class-form.component.ts:374]: I can change it. I think I copied from the js-lib example 🙈

Comment (kilchenmann) [reply src/app/project/ontology/resource-class-form/resource-class-form.component.ts:374]: It's the `UpdateOntology` object. And we need an `UpdateOntology` object for the `label` and one for the `comment`. So, it's an ontology for label (`onto4label`) and an ontology for comment (`onto4comment`)

File: src/app/project/ontology/resource-class-form/resource-class-form.component.ts (hunk 2 above)
Comment (kilchenmann) [reply src/app/project/ontology/resource-class-form/resource-class-form.component.ts:398]: resolved in 076e0a7

**Review by mdelez — APPROVED** [review]

**Pull request merged**